torch/csrc/jit/mobile/type_parser.cpp

```diff
@@ -31,16 +38,30 @@ class TypeParser {
     lex();
   }
 
-  TypePtr parse() {
-    std::string token = next();
-    auto simpleTypeIt = string_to_type_lut().find(token);
-    if (simpleTypeIt != string_to_type_lut().end()) {
-      if (cur() != "]" && cur() != "," && cur() != "") {
-        TORCH_CHECK(
-            false, "Simple type ", token, " is followed by ", "invalid chars.");
-      }
-      return simpleTypeIt->second;
-    } else if (token == "List") {
+  // The list of non-simple types supported by currrent parser.
```

Comment:
Could you run a size test to record how much size aggression?

Reply:
Yeah for ios,
Compressed : +3.8 KiB (+0.01%) => 73.69 MiB
Uncompressed: +8.7 KiB (+0.01%) => 161.83 MiB